llvm/lib/Analysis/CtxProfAnalysis.cpp

```diff
@@ -81,7 +83,55 @@ CtxProfAnalysis::Result CtxProfAnalysis::run(Module &M,
                              toString(MaybeCtx.takeError()));
     return {};
   }
-  return Result(std::move(*MaybeCtx));
+
+  PGOContextualProfile Result;
+
+  for (const auto &F : M) {
+    if (F.isDeclaration())
+      continue;
+    auto GUID = AssignUniqueIDPass::getGUID(F);
+    assert(GUID);
```

Comment:
Maybe `assert(GUID && "Guid not found ....");` otherwise this just looks like `assert(0)` in the failure message.

Reply:
done